container/container_unix.go

```diff
@@ -302,6 +302,9 @@ func (container *Container) UpdateContainer(hostConfig *containertypes.HostConfi
 
 	// update HostConfig of container
 	if hostConfig.RestartPolicy.Name != "" {
+		if container.HostConfig.AutoRemove {
+			return fmt.Errorf("Restart policy of container with AutoRemove flag is not allowed to update")
```

Comment:
Or (given that it's not *updating* that's forbidden, but setting a restart policy in general), perhaps we should try to match the "cli" message; https://github.com/docker/docker/blob/52286b402ef4c25ac90b35f23a9717680e0f8ff7/api/client/container/run.go#L87, something like:


```
Conflicting options: RestartPolicy and AutoRemove
```

Possibly they should be translated back to the CLI error if the daemon returns this? (i.e., to the message set in https://github.com/docker/docker/blob/52286b402ef4c25ac90b35f23a9717680e0f8ff7/api/client/container/run.go#L87)